dashboards/main.go

```diff
@@ -0,0 +1,192 @@
+// This code is heavily --inspired by-- **stolen from** the grafana API examples.
+// https://github.com/grafana-tools/sdk/blob/master/cmd/
+// Rather than separate commands for dashboard and datasources, I have combined the logic into a single
+// file with appropriate changes.
+// dataset files are stored in ./datasources
+// dashboards are stored in ./dashboards
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/gosimple/slug"
+	"github.com/grafana-tools/sdk"
+)
+
+func main() {
+	imode := flag.Bool("import", false, "import dashboards into grafana. (default: false)")
+	grafana := flag.String("grafana", "http://localhost:3000", "Grafana HTTP endpoint")
+	apikey := flag.String("apikey", os.Getenv("GRAFANA_API_KEY"), "Grafana API key")
+	flag.Parse()
+
+	if *grafana == "" {
+		log.Fatal("Unknown grafana API key. Please set GRAFANA_API_KEY or use the -apikey flag")
+	}
+
+	client := sdk.NewClient(*grafana, *apikey, sdk.DefaultHTTPClient)
+
+	// import mode
+	if *imode {
+		ImportDatasources(client)
+		ImportDashboards(client)
+		os.Exit(0)
```

Comment:
putting the backup logic in an `else` block might be a little easier to reason about

Reply:
Thanks. Sounds fine to me -- fixing!